dive_core/perf_metrics_data.cpp

```diff
@@ -0,0 +1,236 @@
+/*
+ Copyright 2025 Google LLC
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+#include "dive_core/perf_metrics_data.h"
+
+#include <algorithm>
+#include <cctype>
+#include <cerrno>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <filesystem>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "dive_core/available_metrics.h"
+
+namespace Dive
+{
+namespace
+{
+constexpr int     kPerfMetricsFixFieldCount = 9;
+const char* const kFixedHeaders[] = { "ContextID",   "ProcessID", "FrameID",
+                                      "CmdBufferID", "DrawID",    "DrawType",
+                                      "DrawLabel",   "ProgramID", "LRZState" };
+
+void trim(std::string& s)
```

Comment:
nit: style-wise, do we prefer `Trim`?
```suggestion
void Trim(std::string& s)
```